src/handler.ts

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IContentsModel, IContentsManager
+} from 'jupyter-js-services';
+
+import * as arrays
+  from 'phosphor-arrays';
+
+import {
+  IMessageFilter, IMessageHandler, Message, installMessageFilter,
+  removeMessageFilter
+} from 'phosphor-messaging';
+
+import {
+  IChangedArgs, Property
+} from 'phosphor-properties';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  Widget, Title
+} from 'phosphor-widget';
+
+import {
+  JupyterCodeMirrorWidget
+} from './widget';
+
+import {
+  loadModeByFileName
+} from './utils';
+
+
+/**
+ * An abstract base class for implementing file handlers.
+ */
+export
+abstract class AbstractFileHandler implements IMessageFilter {
```

Comment:
In a generic app, you'll have the file browser, but not necessarily this
editor.  Also, the generic interface and convenience class should be closer
to where the handler is used.

On Wed, Feb 3, 2016, 20:04 Steven Silvester <notifications@github.com>
wrote:

> In src/handler.ts
> <https://github.com/jupyter/jupyter-js-editor/pull/23#discussion_r51815323>
> :
>
> > +} from 'phosphor-widget';
> > +
> > +import {
> > +  JupyterCodeMirrorWidget
> > +} from './widget';
> > +
> > +import {
> > +  loadModeByFileName
> > +} from './utils';
> > +
> > +
> > +/**
> > + * An abstract base class for implementing file handlers.
> > + */
> > +export
> > +abstract class AbstractFileHandler implements IMessageFilter {
>
> What's the difference between getting it here versus the file browser?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-editor/pull/23/files#r51815323>.
>